icommand.lisp

```diff
@@ -38,22 +38,22 @@
   (message "~s finished" name)
   (pop-top-map))
 
-(defmacro deficommand (name key-bindings &key pre-start-fn cleanup-fn)
+(defmacro deficommand (name (&key bootstrap-fn cleanup-fn) &body key-bindings)
   "Declare an interactive command." ;; TODO
   (let* ((command (if (listp name) (car name) name))
-         (exit-command (concatenate 'string "exit-" (symbol-name command)))
-         (m-name (gensym "m")))
-    `(let ((,m-name (make-sparse-keymap)))
+         (exit-command (concatenate 'string "EXIT-" (symbol-name command)))
+         (keymap (gensym "m")))
+    `(let ((,keymap (make-sparse-keymap)))
        ,@(loop for keyb in key-bindings
-            collect `(define-key ,m-name ,@keyb))
-       (define-key ,m-name (kbd "RET") ,exit-command)
-       (define-key ,m-name (kbd "C-g") ,exit-command)
-       (define-key ,m-name (kbd "ESC") ,exit-command)
+            collect `(define-key ,keymap ,@keyb))
+       (define-key ,keymap (kbd "RET") ,exit-command)
+       (define-key ,keymap (kbd "C-g") ,exit-command)
+       (define-key ,keymap (kbd "ESC") ,exit-command)
        (defcommand ,name () ()
-         ,(concatenate 'string "Starts interactive command \"" ,(symbol-name command) "\"") ;; TODO Add docstring support
-         ,(let ((start-code `(icommand-enter-interactive-mode ,m-name ,(symbol-name command))))
-            (if pre-start-fn
-                `(when (funcall ,pre-start-fn)
+         ,(concatenate 'string "Starts interactive command \"" (symbol-name command) "\"") ;; TODO Add docstring support
```

Comment:
The easiest way to add docstring support would be to use alexandria:parse-body, but we could do that in a subsequent PR.